projects/nlp/unit_test_1.py

```diff
@@ -52,65 +53,63 @@ def instantiateModel(args):
   return model
 
 
-def trainModel(args, model, trainingData, labelRefs):
+def _trainModel(args, model, trainingData, labelRefs):
   """
   Train the given model on trainingData. Return the trained model instance.
   """
   print
-  print "=======================Training model on sample text================"
-  for i, doc in enumerate(trainingData):
-    document = doc[0]
-    labels = doc[1]
-    docId = doc[2]
-    print
-    print "Document=", document, "label=",labelRefs[doc[1][0]], "id=",docId,
+  print "======================Training model on sample text==================="
+  if args.verbosity > 0:
+    printTemplate = "{0:<85}|{1:<10}|{2:<5}"
+    print printTemplate.format("Document", "Label", "ID")
+  for (document, labels, docId) in trainingData:
+    if args.verbosity > 0:
+      print printTemplate.format(
+        wrapper.fill(document), labelRefs[labels[0]], docId)
     model.trainDocument(document, labels, docId)
 
   return model
 
 
-def testModel(args, model, testData, labelRefs, documentCategoryMap):
+def _testModel(args, model, testData, labelRefs, documentCategoryMap):
   """
-  Test the given model on testData and print out accuracy.
+  Test the given model on testData, print out and return results metrics.
 
-  Accuracy is calculated as follows. Each token in a document votes for a single
-  category. The document is classified with the category that received the most
-  votes. Note that it is possible for a token and/or document to receive no
-  votes, in which case it is counted as a misclassification.
+  For each data sample in testData the model infers the similarity to each other
+  sample. From a list sorted most-to-least similar, we then get the ranks of the
+  samples that share the same category as the inference sample. Ideally these
+  ranks would be low. The returned metrics are the min, mean, and max ranks of
+  the category samples.
   """
   print
-  print "==========================Classifying sample text================"
-  numCorrect = 0
-  for recordNum, doc in enumerate(testData):
-    document = doc[0]
-    desiredLabels = doc[1]
+  print "========================Testing on sample text========================"
+  totalScore = 0
+  for i, (document, labels, docId) in enumerate(testData):
+    _, sortedIds, _ = model.inferDocument(
+      document, returnDetailedResults=True, sortResults=True)
+
+    # Compute the test metrics for this document
+    expectedCategory = docId / 100
+    ranks = numpy.array(
```

Comment:
I don't think this is what Jeff meant. He was more precise in the test. For any given category, Sentence 2 has to be more similar to sentence 1 than Sentence 3, and so on.  The sort order has to exactly match the order of the sentences.